R/importFilterHypermutators.R

```diff
@@ -0,0 +1,201 @@
+#' @param dOut Directory to store output, defaults to getwd().
+#' @param xS Mutation threshold. 400 by default.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @param writeDetailedLog Flag for extra details about log. TRUE by default.
+#'
+#' @examples
+#' \dontrun{
+#'     importFilterHypermutators(rSNVFileIn, rCNAFileIn, dOut, xS)
+#' }
+#'
+#' @export
+importFilterHypermutators <- function(rSNVFileIn = c(),
+                                      rCNAFileIn = c(),
+                                      dOut = getwd(),
+                                      xS = 400,
+                                      silent = FALSE,
+                                      writeLog = TRUE,
+                                      writeDetailedLog = TRUE) {
+
+    ## VALIDATE PARAMS ########
+    # make sure that files actually exist
+    for (file in rSNVFileIn) {
+        r <- .checkArgs(file, like = "FILE_E", checkSize = TRUE)
+        if (length(r) > 0) {
+            stop(r)
+        }
+    }
+
+    for (file in rCNAFileIn) {
+        r <- .checkArgs(file, like = "FILE_E", checkSize = TRUE)
+        if (length(r) > 0) {
+            stop(r)
+        }
+    }
+
+    # make sure that output directory is valid
+    r <- .checkArgs(dOut, like = "DIR", checkSize = TRUE)
+    if(length(r) > 0) {
+        stop(r)
+    }
+
+    if (!is.double(xS)) {
+        stop(xS)
+    }
+
+    # set up hashes
+    hashTable <- new.env(hash = TRUE)
+    totalSamples <- 0
+
+    ## COUNT MUTATIONS ########
+    # for each file in rCNAFileIn
+    for (file in rCNAFileIn) {
+        # readRDS, count gene CNAs and add to hash for sample @ CNA and nMUT
+        rCNA <- readRDS(file)
+        for (sample in names(rCNA)) {
+            print(sample)
+            totalSamples <- totalSamples + 1
+            # if key is in hash, increment CNA and nMUT
+            if (!is.null(hashTable$sample)) {
+                prevCNACount <- hashTable$sample$CNA
+                prevTotalCount <- hashTable$sample$total
+                assign(sample, list(CNA = prevCNACount + 1, SNV = 0, total = prevTotalCount + 1), envir = hashTable)
+            }
+            # question - does each gene-CNA count as one mutation? I'm guessing yes?
+            assign(sample, list(CNA = 1, SNV = 0, total = 1), envir = hashTable)
+        }
+    }
+
+    # clean up hashTable
+    assign("Cytoband", NULL, envir = hashTable)
+    assign("Gene.Symbol", NULL, envir = hashTable)
+    assign("Locus.ID", NULL, envir = hashTable)
+
+    # for each file in rSNVFileIn
+    for (file in rSNVFileIn) {
+        # readRDS, increment counter for SNV and nMUT
+        rSNV <- readRDS(file)
+        for (sample in rSNV$Tumor_Sample_Barcode) {
+            # need to substitute dashes with period for consistency in files
```

Comment:
Actually we should oblige the code that imports CNA resp. SNV files to store sample IDs in a consistent format. That's a prime example of an integration problem.